### ISSUE TITLE
What is happening here

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/sherlock#1]
Since this issue is written in the context of the [Monterey framework]()

**Comment by adriatic** [comment]
### Example of a possible benefit from Sherlock
Let's start from the narrowest context - work recently done on **[Aurelia CLI package importer component](https://github.com/monterey-framework/cli)**, briefly defined **[here](https://aurelia-ui-toolkits.gitbooks.io/monterey-developer-manual/content/infrastructure/aurelia_import/design.html)** and expected to be used by the **[Monterey tool](https://aurelia-ui-toolkits.gitbooks.io/monterey-user-manual/content/introduction/what_is_monterey.html)** in the context of creating and managing applications. 

Here is a sketch of the Monterey Package Chooser (MPC) that connect the autocomplete with two simple grids to provide a cascading UI to add packages to the app:

![image](https://cloud.githubusercontent.com/assets/2712405/21534121/317132f4-cd30-11e6-866a-5dba11447daa.png)

- I plan to use **[npms API](https://api-docs.npms.io/)** to feed the autocomplete component (red marker 1) and establish a slick pipeline presenting the developer with all relevant information (to be discussed in more details next). This whole entity should eventually become a part of npms (client side tools).

- I would like to use yarn and work with their team trying to squeeze the size of application's runtime.

- Aurelia CLI component offers a very interesting "abstraction" that will be used to support all module loaders of interest.

In summary, we should see this as an attempt to reach two goals (quoting @JeroenVinke here)

1. reduce the time waiting for npm install
2.  help the user in using a package

**Comment by adriatic** [comment]
### Articles of interest

- [Yarn: A new package manager for JavaScript](https://code.facebook.com/posts/1840075619545360)
